homeassistant/components/climate/fritzhome.py

```diff
@@ -0,0 +1,137 @@
+"""
+Support for AVM Fritz!Box fritzhome thermostate devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/climate.fritzhome/
+"""
+import logging
+
+from homeassistant.components.fritzhome import DOMAIN
+from homeassistant.components.climate import (
+    ATTR_OPERATION_MODE, ClimateDevice, STATE_ECO,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import (PRECISION_HALVES)
+from homeassistant.const import (TEMP_CELSIUS, ATTR_TEMPERATURE)
+
+DEPENDENCIES = ['fritzhome']
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_COMFORT = 'comfort'
+STATE_MANUAL = 'manual'
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE)
+
+OPERATION_LIST = [STATE_COMFORT, STATE_ECO, STATE_MANUAL]
+
+MIN_TEMPERATURE = 8
+MAX_TEMPERATURE = 28
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Fritzhome thermostat platform."""
+    device_list = hass.data[DOMAIN]
+
+    devices = []
+    for device in device_list:
+        if device.has_thermostat:
+            devices.append(FritzhomeThermostat(hass, device))
+
+    add_devices(devices)
+
+
+class FritzhomeThermostat(ClimateDevice):
+    """The thermostat class for Fritzhome thermostates."""
+
+    def __init__(self, hass, device):
+        """Initialize the thermostat."""
+        self._device = device
+        self._current_temperature = self._device.actual_temperature
```

Comment:
Are these values already available at this point? If so, some I/O must have happened at this point, right? It would be preferable to delay the I/O until after the platform has been set up: Set the values to `None` here, and call `add_devices` above with `True` as second parameter. This way, HA will schedule a call to your `update` method instantly.

Reply:
the device with its values is already fetched in the setup of the fritzhome component and the values are valid. So I think this is ok.